legion_core/src/world.rs

```diff
@@ -712,6 +717,56 @@ impl World {
         unsafe { self.get_component_mut_unchecked(entity) }
     }
 
+    pub fn get_all_components<'a, T: Component>(&'a self) -> RefMapSet<'a, Vec<&'a T>> {
```

Comment:
Is this publicly needed?

It will introduce an alternative way to query the world and which 'could' cause confusion.

Reply:
This is #105, the use case I had in mind when adding it was to support imgui.

Reply:
This is now in a separate PR